src/components/Common/TierBlock.js

```diff
@@ -0,0 +1,197 @@
+                          checked={input.value === 'on'}
+                          onChange={() => input.onChange('on')}
+                          value='on'
+                        />
+                        <span className='title'>on</span>
+                      </label>
+                      <label className='radio-inline'>
+                        <input
+                          type='radio'
+                          checked={input.value === 'off'}
+                          value='off'
+                          onChange={() => input.onChange('off')}
+                        />
+                        <span className='title'>off</span>
+                      </label>
+                    </div>
+                    <p className='description'>{DESCRIPTION.ALLOW_MODIFYING}</p>
+                  </div>
+                )}
+              />
+            </div>
+
+            <div className="input-block-container">
+              <Field
+                name={`${name}.startTime`}
+                component={InputField2}
+                validate={validateTierStartDate(index)}
+                errorStyle={inputErrorStyle}
+                type="datetime-local"
+                side="left"
+                label={START_TIME}
+                description={DESCRIPTION.START_TIME}
+              />
+              <Field
+                name={`${name}.endTime`}
+                component={InputField2}
+                validate={validateTierEndDate(index)}
+                errorStyle={inputErrorStyle}
+                type="datetime-local"
+                side="right"
+                label={END_TIME}
+                description={DESCRIPTION.END_TIME}
+              />
+            </div>
+
+            <div className="input-block-container">
+              <Field
+                name={`${name}.rate`}
+                component={InputField2}
+                validate={composeValidators(
+                  isPositive(),
+                  isInteger(),
+                  isLessOrEqualThan('Should not be greater than 1 quintillion (10^18)')('1e18')
+                )}
+                errorStyle={inputErrorStyle}
+                type="text"
+                side="left"
+                label={RATE}
+                description={DESCRIPTION.RATE}
+              />
+              <Field
+                name={`${name}.supply`}
+                component={InputField2}
+                validate={isPositive()}
+                errorStyle={inputErrorStyle}
+                type="text"
+                side="right"
+                label={SUPPLY}
+                description={DESCRIPTION.SUPPLY}
+              />
+              {
+                /*
+                  * TODO: REVIEW. I'm not sure about this approach.
+                  * But it worked for me to keep the error messages properly updated for the minCap field.
+                  */
+              }
+              <Field name="minCap" subscription={{}}>
```

Comment:
Same here thing (as in the Error component). Is the Field necessary? Can't we use the [`change`](https://github.com/final-form/react-final-form#change-name-string-value-any--void) function?